src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -566,4 +568,13 @@ private static void validateSingleQueryType(Integer k, Float distance, Float sco
             throw new IllegalArgumentException(String.format("[%s] requires exactly one of k, distance or score to be set", NAME));
         }
     }
+
+    static <T> void updateQueryStats(T queryParam, QueryBuilder filter, KNNCounter queryCounter, KNNCounter queryWithFilterCounter) {
```

Comment:
why this function is package private?

Reply:
@navneet1v Updated to class scope private